aiohttp/client.py

```diff
@@ -383,7 +384,13 @@ def update_body_from_data(self, data):
             assert not isinstance(data, io.StringIO), \
                 'attempt to send text data instead of binary'
             self.body = data
-            self.chunked = True
+            if not self.chunked and isinstance(data, io.BufferedReader):
+                # Not chunking if content-length can be determined
+                size = fstat(data.fileno()).st_size - data.tell()
```

Comment:
I would use full `os.fstat` here since it doesn't simply to figure out what fstat is and where it comes from. Also look on the other imports from stdlib.